src/glutil.cpp

```diff
@@ -32,61 +47,80 @@ void TransformState::setUniforms(GLuint prog) const
     setUniform(prog, "modelViewProjectionMatrix", mvproj);
 }
 
-
-void TransformState::load() const
+void TransformState::setOrthoProjection(double left, double right, double bottom, double top, double nearVal, double farVal)
 {
-    glMatrixMode(GL_PROJECTION);
-    glLoadMatrix(M44f(projMatrix));
-    glMatrixMode(GL_MODELVIEW);
-    glLoadMatrix(M44f(modelViewMatrix));
-}
+    // Note: Create an orthographic matrix from input values
+    if ((left == right) || (bottom == top) || (nearVal == farVal))
+        throw DisplazError("Invalid input for orthographic projection.");
+
+    double xx = 2.0/(right-left);
+    double yy = 2.0/(top-bottom);
+    double zz = -2.0/(farVal-nearVal);
+
+    double tx = -(right+left)/(right-left);
+    double ty = -(top+bottom)/(top-bottom);
+    double tz = -(farVal+nearVal)/(farVal-nearVal);
 
+    projMatrix = M44d(xx,0.0,0.0,0.0,
+                      0.0,yy,0.0,0.0,
+                      0.0,0.0,zz,0.0,
+                      tx,ty,tz,1.0);
+}
 
 //------------------------------------------------------------------------------
 
-void drawBoundingBox(const TransformState& transState,
-                     const Imath::Box3d& bbox,
-                     const Imath::C3f& col)
+void drawBox(const TransformState& transState,
+             const Imath::Box3d& bbox,
+             const Imath::C3f& col,
+             const GLuint& shaderProgram)
 {
     // Transform to box min for stability with large offsets
+    // This function is allowing you to render any box, not necessarily having anything to do
+    // with the bounding box (e.g. hcloudview is using this in drawBounds)
```

Comment:
Sure, no problems renaming it for clarity.

A quibble about this kind of comment - someone reading it later won't have the context of knowing the function was originally used to draw a bounding box (or was indeed called `drawBoundingBox()`), so talking about bounding boxes here will probably just confuse them.  I try to avoid comments which can only be understood in the context of the history of the codebase.